streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -685,13 +685,13 @@ public boolean test(final K1 key, final V1 value) {
                                                                                     windows.maintainMs());
 
             KStreamKStreamJoin<K1, R, V1, V2> joinThis = new KStreamKStreamJoin<>(otherWindow.name(),
-                                                                                  windows.before,
                                                                                   windows.after,
+                                                                                  windows.before,
```

Comment:
Your SQL formulation is incorrect.  E.g. for a record of in `stream1` with a timestamp of `1000`, with a `before` of `100` and an `after` of `0`, your formulation would result in:

```
stream1.ts - before >= stream2.ts AND stream1.ts + after <= stream2.ts
1000 - 100 >= stream2.ts AND 1000 + 0 <= stream2.ts
900 >= stream2.ts AND 1000 <= stream2.ts
```
which is equivalent to
```
stream2.ts <= 900 AND stream2.ts >= 1000
```
which is self-evidently always false.

The correct formulation is:

```sql
SELECT * FROM stream1, stream2
WHERE
  stream1.key = stream2.key
  AND
  stream2.ts >= stream1.ts - before
  AND
  stream2.ts <= stream1.ts + after
```